zenith_utils/src/pq_proto.rs

```diff
@@ -37,19 +37,18 @@ pub enum FeMessage {
 }
 
 #[derive(Debug)]
-pub struct FeStartupMessage {
-    pub version: u32,
-    pub kind: StartupRequestCode,
-    // optional params arriving in startup packet
-    pub params: HashMap<String, String>,
-}
-
-#[derive(Debug)]
-pub enum StartupRequestCode {
-    Cancel,
-    NegotiateSsl,
-    NegotiateGss,
-    Normal,
+pub enum FeInitialMessage {
+    CancelRequest {
+        backend_pid: i32,
+        cancel_key: i32,
```

Comment:
Do you think these make more sense as i32s or u32s? (Also applies to #1063)

Reply:
Postgres keeps them in i32 so I'm copying that. For cancel_key it technically doesn't matter since we're only sending/receiving it in binary, but the pid we get from a pg query so it matters.